src/synapse/workload.py

```diff
@@ -673,8 +673,10 @@ def get_environment(charm_state: CharmState) -> typing.Dict[str, str]:
         A dictionary representing the Synapse environment variables.
     """
     environment = {
-        "SYNAPSE_SERVER_NAME": f"{charm_state.synapse_config.server_name}",
+        "SYNAPSE_CONFIG_DIR": SYNAPSE_CONFIG_DIR,
+        "SYNAPSE_CONFIG_PATH": SYNAPSE_CONFIG_PATH,
         "SYNAPSE_REPORT_STATS": f"{charm_state.synapse_config.report_stats}",
+        "SYNAPSE_SERVER_NAME": f"{charm_state.synapse_config.server_name}",
```

Comment:
Maybe also adding a new variable `SYNAPSE_DATA_DIR`? That one is currently is '/data' in the `start.py` script that creates the initial configuration and in `configure_workers_and_start.py`. It is kind of implicit now....